Bearded.UI/Events/EventRouter.cs

```diff
@@ -8,25 +8,19 @@ namespace Bearded.UI.Events
 {
     static class EventRouter
     {
-        public delegate void RoutedEvent<in T>(Control control, T eventData) where T : RoutedEventArgs;
-
-        public static EventPropagationPath FindPropagationPath(IControlParent root, Predicate<Control> childSelector)
+        public enum PropagationTestOutcome
         {
-            var parent = root;
-            var path = new List<Control>();
-
-            while (parent != null)
-            {
-                var child = parent.Children.Reverse().FirstOrDefault(childSelector.Invoke);
-                if (child != null)
-                {
-                    path.Add(child);
-                }
+            Failure,
+            PassThrough,
+            Success
```

Comment:
Wouldn't 'Hit' and 'Miss' be more meaningful than the more generic 'Success' and 'Failure'?